e2e/test/PoolingOverAmqp.cs

```diff
@@ -65,90 +92,123 @@ public static class PoolingOverAmqp
                     TestDevice testDevice = await TestDevice.GetTestDeviceAsync($"{devicePrefix}_{i}_").ConfigureAwait(false);
                     DeviceClient deviceClient = testDevice.CreateDeviceClient(transportSettings, authScope);
 
-                    var amqpConnectionStatusChange = new AmqpConnectionStatusChange();
+                    var amqpConnectionStatusChange = new AmqpConnectionStatusChange(devicePrefix);
                     deviceClient.SetConnectionStatusChangesHandler(amqpConnectionStatusChange.ConnectionStatusChangesHandler);
 
                     testDevices.Add(testDevice);
                     deviceClients.Add(deviceClient);
                     amqpConnectionStatuses.Add(amqpConnectionStatusChange);
 
-                    operations.Add(initOperation(deviceClient, testDevice));
+                    operations.Add(deviceClient.OpenAsync());
                 }
                 await Task.WhenAll(operations).ConfigureAwait(false);
                 operations.Clear();
 
                 try
                 {
-                    for (int i = 0; i < devicesCount; i++)
-                    {
-                        operations.Add(testOperation(deviceClients[i], testDevices[i]));
-                    }
-                    await Task.WhenAll(operations).ConfigureAwait(false);
-                    operations.Clear();
-
                     // Close the device client instances and verify the connection status change checks
-                    bool deviceConnectionStatusAsExpected = true;
                     for (int i = 0; i < devicesCount; i++)
                     {
                         await deviceClients[i].CloseAsync().ConfigureAwait(false);
 
                         // The connection status change count should be 2: connect (open) and disabled (close)
-                        if (amqpConnectionStatuses[i].ConnectionStatusChangesHandlerCount != 2)
-                        {
-                            deviceConnectionStatusAsExpected = false;
-                        }
+                        Assert.AreEqual(2, amqpConnectionStatuses[i].ConnectionStatusChangeCount);
 
                         // The connection status should be "Disabled", with connection status change reason "Client_close"
                         Assert.AreEqual(ConnectionStatus.Disabled, amqpConnectionStatuses[i].LastConnectionStatus, $"The actual connection status is = {amqpConnectionStatuses[i].LastConnectionStatus}");
                         Assert.AreEqual(ConnectionStatusChangeReason.Client_Close, amqpConnectionStatuses[i].LastConnectionStatusChangeReason, $"The actual connection status change reason is = {amqpConnectionStatuses[i].LastConnectionStatusChangeReason}");
+
+                        reRunTest = false;
```

Comment:
reRunTest is never set as true; it should be set to true in the catch block (when Exception is AssertionFailedException)

Reply:
You are absolutely right, I'll add it